pkg/local/diff.go

```diff
@@ -202,20 +202,19 @@ func WalkS3(root string, callbackFunc func(p string, info fs.FileInfo, err error
 		}
 		if info.IsDir() {
 			// TODO: We don't return dir results for the listing, how will this effect directory markers, and can we even support directory markers?
-			// Save encountered directories in a sorted queue and compare them with the first appearance of an file in that level
-			heap.Push(&dirQueue, p+path.Separator) // add path separator to dir name and sort it later
+			// Save encountered directories in a min heap and compare them with the first appearance of a file in that level
+			heap.Push(&stringHeap, p+path.Separator) // add path separator to dir name and sort it later
 			return filepath.SkipDir
 		}
 
-		heap.Init(&dirQueue)
-		for dirQueue.Len() > 0 {
-			dir := heap.Pop(&dirQueue).(string)
+		for stringHeap.Len() > 0 {
+			dir := stringHeap.Peek().(string)
 			if p > dir {
+				heap.Pop(&stringHeap) // remove from queue
```

Comment:
We can't because it is the heap's Pop and Push logic that actually re-organizes the tree as part of the operation. the stringHeap object itself is rather "dumb".
We have to maintain the interface which returns `any`, otherwise we could not use the heap functionality

Reply:
See https://pkg.go.dev/container/heap#Interface